clang/lib/StaticAnalyzer/Checkers/CheckSecuritySyntaxOnly.cpp

```diff
@@ -754,7 +760,22 @@ void WalkAST::checkDeprecatedOrUnsafeBufferHandling(const CallExpr *CE,
   if (!filter.check_DeprecatedOrUnsafeBufferHandling)
     return;
 
-  if (!BR.getContext().getLangOpts().C11)
+  const bool ShouldReport = [this] {
+    const bool IsAnnexKAvailable = analysis::isAnnexKAvailable(
+        &BR.getPreprocessor(), BR.getContext().getLangOpts());
+    const bool IsC11OrLaterStandard = BR.getContext().getLangOpts().C11;
```

Comment:
Do we cache this, or its gonna trigger every time we call `checkDeprecatedOrUnsafeBufferHandling`?
Would that happen only once?